Source/WebInspectorUI/UserInterface/Controllers/WebInspectorExtensionController.js

```diff
@@ -125,15 +125,32 @@ WI.WebInspectorExtensionController = class WebInspectorExtensionController exten
         return {"result": extensionTabID};
     }
 
-    evaluateScriptForExtension(extensionID, scriptSource, {frameURL, contextSecurityOrigin, useContentScriptContext} = {})
+    async evaluateScriptForExtension(extensionID, scriptSource, {frameURL, contextSecurityOrigin, useContentScriptContext} = {})
     {
         let extension = this._extensionForExtensionIDMap.get(extensionID);
         if (!extension) {
             WI.reportInternalError("Unable to evaluate script for extension with unknown ID: " + extensionID);
             return WI.WebInspectorExtension.ErrorCode.InvalidRequest;
         }
 
-        let frame = this._frameForFrameURL(frameURL);
+        let retryUntilTruthy = async (fn, {delay, retries} = {}) => {
+            retries ??= 30;
+            delay ??= 100;
+            let count = 0;
+
+            while (count < retries) {
+                count++;
+                let result = fn();
+                if (result)
+                    return result;
+
+                await Promise.delay(delay);
+            }
+
+            return null;
+        }
```

Comment:
This seems like a useful thing to put in Base as a standalone helper, like `Debouncer` and `Throttler`.

Reply:
It's a tiny utility, so perhaps it can go in [Source/WebInspectorUI/UserInterface/Base/Utilities.js](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/UserInterface/Base/Utilities.js) instead of having its own dedicated file.